### PR TITLE
implement multiple generics

### PR DESCRIPTION
**Pull request by danieldaeschle**
basic example works. some cases needs fixes. help is welcomed. there are many edge cases which need tests.

### CODE REVIEW

**Comment by danieldaeschle** [comment]
failing vls is expected. it needs to be fixed after merging this.

File: vlib/v/ast/ast.v [hunk 1]
```diff
@@ -336,6 +336,11 @@ pub mut:
 	scope         &Scope
 }
 
+pub struct GenericParam {
+pub:
+	name string
```
[the remark this answers is not in the document]
Comment (danieldaeschle) [reply vlib/v/ast/ast.v:341]: maybe not needed for now.

**Comment by danieldaeschle** [comment]
@Larpon why is vab failing?

**Comment by nedpals** [comment]
noted 🙂

**Review by spytheman — APPROVED** [review]
Good work!

**Comment by danieldaeschle** [comment]
Created PR for vls: https://github.com/vlang/vls/pull/63

**Pull request merged**